lib/service/service.go

```diff
@@ -1053,13 +1052,6 @@ func waitAndReload(ctx context.Context, sigC <-chan os.Signal, cfg servicecfg.Co
 func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 	var err error
 
-	// auth and proxy benefit from precomputing keys since they can experience spikes in key
-	// generation due to web session creation and recorded session creation respectively.
-	// for all other agents precomputing keys consumes excess resources.
-	if cfg.Auth.Enabled || cfg.Proxy.Enabled {
-		native.PrecomputeKeys()
-	}
```

Comment:
I have switched to lazily starting the RSA key precomputation where we actually generate keys for web sessions or proxy-recorded sesssions/agentless sessions. It will now only start the first time we generate an RSA key, which will be *never* for the newer algorithm suites and should be acceptable even when using the legacy suite. Importantly, most tests will now never start the RSA key precomputation which will be a huge win for saving CPU time.

Reply:
yes, because now we exclusively start the key precomputation closer to where the keys are actually needed, in newWebSession or generateHostCert, which will only be called on auth or proxy